common/Geometry2d/Pose.hpp

```diff
@@ -128,12 +128,38 @@ class Pose {
         return *this;
     }
 
+    /**
+     * Equality comparison operation.
+     */
+    bool operator==(const Pose& other) const {
+        return _position == other._position && _heading == other._heading;
+    }
+
+    /**
+     * Inequality comparison operation.
+     */
+    bool operator!=(const Pose& other) const {
+        return _position != other._position || _heading != other._heading;
```

Comment:
Since you have the equality operator, you can just do `!(*this == other)`. This reduces copy paste errors